### ISSUE TITLE
UnsafeCallOnNullableType not reporting !! usage

### ISSUE THREAD

**Issue by lyrafelipe** [detekt/detekt#2036]
## Expected Behavior
`UnsafeCallOnNullableType` rule should report usages of `!!` as an unsafe call on nullable type.

## Observed Behavior
`UnsafeCallOnNullableType` rule isn't reporting `!!` as an unsafe call on nullable type after 1.1.0 update.

## Steps to Reproduce
When running the following test:

```kotlin
class UnsafeCallOnNullableTypeTest {
    private val code: String =
            """
                val aString: String?
                aString = null
                val aStringLength = aString!!.length
            """.trimIndent()

    @Test
    fun shouldReportDoubleBangUsage() {
        val findings = UnsafeCallOnNullableType().lint(code)

        assertThat(findings).hasSize(1)
        assertThat(findings[0].message).isEqualTo("Calling !! on a nullable type will throw a " +
                "NullPointerException at runtime in case the value is null. It should be avoided.")
    }
}
```
The test won't pass if the `UnsafeCallOnNullableType` rule after f80fe60383c78bed811cdad1fd54402e21b58a11 commit is used but, if you remove the 3 lines added to the rule by the linked commit or use version `1.0.1`, it will pass.

## Your Environment

* Version of detekt used: 1.1.1
* Version of Gradle used (if applicable): 5.6.2
* Operating System and version: Ubuntu Mate 18.04
* Link to your project (if it's a public repository): Private Repository

**Comment by 3flex** [comment]
The behaviour of the rule has changed to now require type resolution, which means if you're running detekt without type resolution (not passing a classpath) then it won't run the rule.

We need docs for this and have an issue open already to cover that.

If you're actually running detekt with type resolution and the rule is still not picking up this case then we'll need to look more closely at the problem.

**Comment by lyrafelipe** [comment]
By passing a classpath you mean running detekt with `--classpath` flag? If yes, could you link an example?

**Comment by 3flex** [comment]
One way would be running `gradlew detektMain` with debug output and looking for the flags passed to detekt when it runs.

**Comment by lyrafelipe** [comment]
I ran the task `gradlew detekt` since I didn't have the task `detektMain` with debug output and couldn't find any `classpath` flag being passed to detekt.

Could you please clarify how should I do this? Is there a way to declare it inside my `detekt` block like `toolVersion`, `input`, filters, etc?

**Comment by 3flex** [comment]
Gradle creates tasks for each source set but currently only on Kotlin JVM projects, is it maybe an Android, native or JS project?

We definitely need docs for this so probably creating those is the action we need to take to resolve this, see #1943

**Comment by lyrafelipe** [comment]
It's an Android project.

**Comment by 3flex** [comment]
Yeah we're not generating required Gradle tasks for Android yet. It's on my to do list; I'll work on it next since there's a clear need.

**Comment by schalkms** [comment]
The required Gradle tasks for Android are described in another issue. Therefore, I’ll close this.

**Comment by lyrafelipe** [comment]
@schalkms Can you link the issue with this TODO?

**Comment by schalkms** [comment]
Link #2259
